File: gitdot-api-ts/src/endpoint/migration/list-migrations.ts
import { z } from "zod";

import { MigrationResource } from "../../resource";

export const ListMigrationsRequest = z.object({});
export type ListMigrationsRequest = z.infer<typeof ListMigrationsRequest>;

export const ListMigrations = {
  path: "/migrations",
  method: "GET",
  request: ListMigrationsRequest,
  response: z.array(MigrationResource),
} as const;
export type ListMigrations = typeof ListMigrations;

File: gitdot-api/src/endpoint/migration/list_migrations.rs
use crate::{endpoint::Endpoint, resource::migration::MigrationResource};

pub struct ListMigrations;

impl Endpoint for ListMigrations {
    const PATH: &'static str = "/migrations";
    const METHOD: http::Method = http::Method::GET;

    type Request = ();
    type Response = ListMigrationsResponse;
}

pub type ListMigrationsResponse = Vec<MigrationResource>;
